autogpt_platform/frontend/src/lib/supabase/hooks/useSupabase.ts

```diff
@@ -0,0 +1,163 @@
+"use client";
+import { useEffect, useMemo, useState, useRef } from "react";
+import { createBrowserClient } from "@supabase/ssr";
+import { SignOut, User } from "@supabase/supabase-js";
+import { useRouter } from "next/navigation";
+import {
+  broadcastLogout,
+  getRedirectPath,
+  isLogoutEvent,
+  setupSessionEventListeners,
+} from "../helpers";
+
+export function useSupabase() {
```

Comment:
Mostly what was [on the original hook](https://github.com/Significant-Gravitas/AutoGPT/pull/10150/files#diff-914c55c53e4342d3e4796f15004e9de8e3d60bb66dc78bf2d0e30d7f96e8faa8) + the cross tab session sync.